### ISSUE TITLE
Query Relationships on Materialized Path Tree Not Working

### ISSUE THREAD

**Issue by mattsidjohn** [nestjsx/crud#419]
query: {
    join: {
      conversationParticipants: {
        eager: true,
      },
      children:{
        eager: true,
      },
      'children.conversationParticipants': {
        eager: true
      },
    },
  },

Returns error

{
    "message": "table name \"conversationParticipants\" specified more than once",
    "statusCode": 400
}

**Comment by zMotivat0r** [comment]
you need to set `alias` either for `conversationParticipants` or for `children.conversationParticipants`

**Comment by mattsidjohn** [comment]
@zMotivat0r I had tried that and got this error.

`@Crud({
  model: { type: Conversation },

  params: {
    id: {
      field: 'id',
      primary: true,
      type: 'uuid',
    },
  },
  query: {
    join: {
      conversationParticipants: {
        eager: true,
      },
      children: {
        eager: true,
      },
      'children.conversationParticipants': {
        eager: true,
        alias: 'participants',
      }
    },
  },
})`

`src/conversations/conversations.controller.ts:28:9 - error TS2322: Type '{ join: { conversationParticipants: { eager: boolean; }; children: { eager: boolean; }; 'children.conversationParticipants': { eager: boolean; alias: string; }; }; }' is not assignable to type 'QueryOptions'.
  Types of property 'join' are incompatible.
    Type '{ conversationParticipants: { eager: boolean; }; children: { eager: boolean; }; 'children.conversationParticipants': { eager: boolean; alias: string; }; }' is not assignable to type 'JoinOptions'.
      Property ''children.conversationParticipants'' is incompatible with index signature.
        Type '{ eager: boolean; alias: string; }' is not assignable to type 'JoinOption'.
          Object literal may only specify known properties, and 'alias' does not exist in type 'JoinOption'.

28         alias: 'participants',
           ~~~~~~~~~~~~~~~~~~~~~

  node_modules/@nestjsx/crud/lib/interfaces/crud-options.interface.d.ts:7:5
    7     query?: QueryOptions;
          ~~~~~
    The expected type comes from property 'query' which is declared here on type 'CrudOptions'

    at createTSError (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:421:12)
    at reportTSError (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:425:19)
    at getOutput (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:530:36)
    at Object.compile (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:735:32)
    at Module.m._compile (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:814:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:973:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/mattsidjohn/Projects/gloo-mwi-repo/conversations-service/node_modules/ts-node/src/index.ts:817:12)
    at Module.load (internal/modules/cjs/loader.js:812:32)
    at Function.Module._load (internal/modules/cjs/loader.js:724:14)
    at Module.require (internal/modules/cjs/loader.js:849:19)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.`

**Comment by zMotivat0r** [comment]
what version of the lib do you use?

**Comment by zMotivat0r** [comment]
fixed in [v4.6.0](https://github.com/nestjsx/crud/releases/tag/v4.6.0)
